sentry/const.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 Versada <https://versada.eu/>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import collections
+import logging
+
+import odoo.loglevels
+
+_logger = logging.getLogger(__name__)
+try:
+    import raven
+    from raven.conf import defaults
+except ImportError:
+    _logger.debug('Cannot import "raven". Please make sure it is installed.')
+
+
+def split_multiple(string, delimiter=',', strip_chars=None):
+    '''Splits :param:`string` and strips :param:`strip_chars` from values.'''
+    if not string:
+        return []
+    return [v.strip(strip_chars) for v in string.split(delimiter)]
+
+
+SentryOption = collections.namedtuple(
+    'SentryOption', ['key', 'default', 'converter'])
+
+# Mapping of Odoo logging level -> Python stdlib logging library log level.
+LOG_LEVEL_MAP = dict([
+    (getattr(odoo.loglevels, 'LOG_%s' % x), getattr(logging, x))
+    for x in ('CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET')
+])
+DEFAULT_LOG_LEVEL = 'warn'
+
+DEFAULT_TRANSPORT = 'threaded'
+TRANSPORT_CLASS_MAP = {
+    'requests_synchronous': raven.transport.RequestsHTTPTransport,
+    'requests_threaded': raven.transport.ThreadedRequestsHTTPTransport,
+    'synchronous': raven.transport.HTTPTransport,
+    'threaded': raven.transport.ThreadedHTTPTransport,
+}
+
+ODOO_USER_EXCEPTIONS = [
+    'odoo.exceptions.AccessDenied',
+    'odoo.exceptions.AccessError',
+    'odoo.exceptions.MissingError',
+    'odoo.exceptions.RedirectWarning',
+    'odoo.exceptions.UserError',
+    'odoo.exceptions.ValidationError',
+    'odoo.exceptions.Warning',
+    'odoo.exceptions.except_orm',
```

Comment:
Odoo uses these exceptions as part of flow control and these exceptions are integrated more nicely with the client, eg. in Odoo 8, if you try to add a line to a SO before setting the partner in the SO form, you will get an `except_orm` informing about the partner not being set. I think getting such exceptions in Sentry would cause more noise than it would do good, so these are ignored by default (but, as you've noticed, it's configurable with `sentry_ignored_exceptions`).